### PR TITLE
Update how study view sorts the clinical attributes

### PR DESCRIPTION
**Pull request by zhx828**
# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/5815  
https://github.com/cBioPortal/cbioportal/issues/3800

### CODE REVIEW

File: src/config/serverConfigDefaults.ts [hunk 1]
```diff
@@ -90,21 +90,40 @@ const ServerConfigDefaults: Partial<IServerConfig> = {
         tableAttrs: ['SAMPLE_CANCER_TYPE', 'SAMPLE_CANCER_TYPE_DETAILED'],
         priority: {
             "SAMPLE_CANCER_TYPE": 3000,
+            "PATIENT_CANCER_TYPE": 3000,
```
Comment (zhx828) [inline src/config/serverConfigDefaults.ts]: Please ignore these duplicate default settings. These will be cleaned up once the backend does not require to define which group(sample/patient) the clinical attribute in. For now, we need to define them separately due the the same attribute could be defined as patient/sample differently on study level

File: src/pages/studyView/StudyViewUtils.tsx [hunk 2]
```diff
@@ -1322,12 +1317,31 @@ export function clinicalDataCountComparator(a: ClinicalDataCount, b: ClinicalDat
...
+        displayName: a.displayName
+    }, {
+        priority: Number(b.priority),
+        displayName: b.displayName
+    });
+}
+
+// Descent sort priority then ascent sort by display name
+export function chartMetaComparator(a: ChartMeta, b: ChartMeta): number {
+    return clinicalAttributeSortingComparator(_.pick(a, ['priority', 'displayName']), _.pick(b, ['priority', 'displayName']))
```
Comment (alisman) [inline src/pages/studyView/StudyViewUtils.tsx]: not typesafe b/c props passed as string.   porbably just just do   ` {  priority:a.priority, displayName:a.displayName  } `
but actually, why not just pass the whole object?  why do you need to get rid of the other things?

Comment (zhx828) [reply src/pages/studyView/StudyViewUtils.tsx]: The function is used in two places but the models are different. They just have common properties priority and displayName which are only required component.

**Comment by zhx828** [comment]
@alisman this pr is updated

**Pull request merged**